src/System.IO.FileSystem/tests/Enumeration/RemovedDirectoryTests.netcoreapp.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO.Enumeration;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Tests.Enumeration
+{
+    public class RemovedDirectoryTests : FileSystemTest
+    {
+        private class DirectoryFinishedEnumerator : FileSystemEnumerator<string>
+        {
+            public DirectoryFinishedEnumerator(string directory, EnumerationOptions options)
+                : base(directory, options)
+            {
+            }
+
+            protected override string TransformEntry(ref FileSystemEntry entry)
+            {
+                return entry.FileName.ToString();
+            }
+
+            public delegate void DirectoryFinishedDelegate(ReadOnlySpan<char> directory);
```

Comment:
FWIW, this could use:
https://github.com/dotnet/coreclr/blob/a8d40ac8362bacd98f7fa422190fe66916e040e8/src/System.Private.CoreLib/shared/System/Action.cs#L37

Reply:
Cool, thanks!

Reply:
I'm a little confused by the ReadOnlySpanAction<T, TArg> signature. Why is there T and TArg? We're always presuming two arguments? Why didn't we have ReadOnlySpanAction<T> as well (i.e., what I would use here presumably)?

Reply:
> We're always presuming two arguments?

It's intended for perf-sensitive APIs, in which case the majority of cases where you're reading from the span you'd be storing the data somewhere, and in that case you'd be storing it to an argument rather than to something you close over, which would force allocations.  So we're presuming at least one argument in addition to the span, but that argument can be a tuple, so you can effectively pass in any positive number of arguments.

Reply:
> we're presuming at least one argument in addition to the span, but that argument can be a tuple, so you can effectively pass in any positive number of arguments.

Thanks for the clarification.